app/controllers/rentals_controller.rb

```diff
@@ -0,0 +1,89 @@
+class RentalsController < ApplicationController
+  
+  def check_out     
+    customer = Customer.find_by(id: params[:customer_id])
+    if customer.nil?
+      render json: { errors: ['Not Found'] }, status: :not_found
+      return
+    end
+    
+    video = Video.find_by(id: params[:video_id])
+    
+    if video.nil?
+      render json: { errors: ['Not Found'] }, status: :not_found
+      return
+    end
+    
+    if video[:available_inventory] == 0
+      render json: { errors: 'Video not available' }, status: :bad_request
+      return
+    end
+    
+    rental = Rental.new(customer_id: customer.id, video_id: video.id, due_date: Date.today + 7)
+    
+    if rental.save
+      customer[:videos_checked_out_count] += 1
+      customer.save
+      
+      video[:available_inventory] -= 1
+      video.save
+      
+      render json: { 
+        due_date: rental.due_date, 
+        customer_id: customer.id, 
+        video_id: video.id, 
+        videos_checked_out_count: customer.videos_checked_out_count, 
+        available_inventory: video.available_inventory }, status: :ok
+        return
+        
+      else
+        render json: { errors: rental.errors.full_messages }, status: :bad_request
+        return 
+      end 
```

Comment:
There doesn't seem to be to be a way for the user's request to be bad at this point because the customer_id and the video_id have already both been verified. 
Because of that, it could still be appropriate to have an else clause here but the only way it'd get executed is if the connection to the database failed. In that case, the more appropriate status would be an `:internal_server_error`.

Reply:
Please let me know if this comment is unclear!